pkg/services/cloudmigration/api/api.go

```diff
@@ -40,19 +40,28 @@ func RegisterApi(
 // registerEndpoints Registers Endpoints on Grafana Router
 func (cma *CloudMigrationAPI) registerEndpoints() {
 	cma.routeRegister.Group("/api/cloudmigration", func(cloudMigrationRoute routing.RouteRegister) {
+		// destination instance endpoints for token management
+		cloudMigrationRoute.Get("/token", routing.Wrap(cma.GetToken))
+		cloudMigrationRoute.Post("/token", routing.Wrap(cma.CreateToken))
+		cloudMigrationRoute.Delete("/token/:uid", routing.Wrap(cma.DeleteToken))
+
+		// on-prem instance endpoints for managing GMS sessions
 		cloudMigrationRoute.Get("/migration", routing.Wrap(cma.GetSessionList))
 		cloudMigrationRoute.Post("/migration", routing.Wrap(cma.CreateSession))
 		cloudMigrationRoute.Get("/migration/:uid", routing.Wrap(cma.GetSession))
 		cloudMigrationRoute.Delete("/migration/:uid", routing.Wrap(cma.DeleteSession))
 
-		// TODO new APIs for snapshot management to replace these
+		// sync approach to data migration
 		cloudMigrationRoute.Post("/migration/:uid/run", routing.Wrap(cma.RunMigration))
 		cloudMigrationRoute.Get("/migration/:uid/run", routing.Wrap(cma.GetMigrationRunList))
 		cloudMigrationRoute.Get("/migration/run/:runUID", routing.Wrap(cma.GetMigrationRun))
 
-		cloudMigrationRoute.Get("/token", routing.Wrap(cma.GetToken))
-		cloudMigrationRoute.Post("/token", routing.Wrap(cma.CreateToken))
-		cloudMigrationRoute.Delete("/token/:uid", routing.Wrap(cma.DeleteToken))
+		// async approach to data migration using snapshots
+		cloudMigrationRoute.Post("/migration/:uid/snapshot", routing.Wrap(cma.CreateSnapshot))
+		cloudMigrationRoute.Get("/migration/:uid/snapshot/:snapshotUid", routing.Wrap(cma.GetSnapshot))
+		cloudMigrationRoute.Get("/migration/:uid/snapshot", routing.Wrap(cma.GetSnapshotList))
```

Comment:
No big deal: I would name is /migration/:uid/snapshot`s` since it returns a list

Reply:
done